### DISCUSSION TITLE
Suggestion: Position PhotoPrism as replacement for Synology Photostation/Moments/Photos

### DISCUSSION THREAD

**Discussion by friesoft** [photoprism/photoprism#1149] [General]
Disclaimer: I'm a long time Synology Photostation user and probably very biased by this workflow/how things work there, but I guess this is the whole point here... Of course you can close this issue as invalid if this doesn't fit into the scope of PhotoPrism.

With the upcoming release of Synology DSM 7 (releasing probably sometime in 2021) the current photo management solutions (Photostation: professional use, Moments: home use) are going to be merged into "Synology Photos" (the worst of both solutions)

Sadly many features of Photostation are going to get lost in the transition, which will leave many frusted Synology users behind looking for an alternative.
See for example:
* https://community.synology.com/enu/forum/20/post/139322
* https://community.synology.com/enu/forum/20/post/139889
* https://community.synology.com/enu/forum/20/post/141583
* https://community.synology.com/enu/forum/20/post/139158

In your faq you are stating that PhotoPrism might only provide a treeview in a professional edition sometime in the future, but in my opinion this is a mistake as this leaves all Synology users behind looking for an alternative.
In our case we have a carefully crafted folder structure (years and below the different events) with about 5000 folders. I guess we are beyond the point of restructuring everything into albums/whatever.

In my opinion the following features are needed to allow a transition from Synology Photostation to PhotoPrism
* Folder view and treeview (context menus with actions like renaming, moving, permissions, ....)
* Metadata Editor (I guess on this PhotoPrism already exceeds all existing solutions)
* Auto Update Index in case photos are moved/renamed/added/deleted outside of PhotoPrism
* Apps for Android/iOS including photo upload
* Apps for the Big Screen (Android TV/Apple TV)
* User Permissions based on folders
* Don't regenerate thumbnails - use the existing once generated by Photostation (probably technically infeasible?)

Don't get me wrong, I really like PhotoPrism, but it would be much more useful to at least me if it would provide a tree and folder view to manage photos right from PhotoPrism. In the faq you are stating that one could use Windows Explorer/Finder/... to do this => yes we are managing our photos using those tools but we want to look at them afterwards using the same carefully crafted folder structure from our phones, tablets, tv, ... We are also using Photostation to share some parts of our folder structure with other parts of the family, so we know that they can see everything in folder xyz and nothing else.

**Comment by graciousgrey** [comment]
Thank you for sharing this! It is always helpful for us to understand your needs :)

Regarding your feature wishlist:
- Folder view and treeview (context menus with actions like renaming, moving, permissions, ....)
=> We already have the folders section (displaying your folders without hierarchy) and the originals section (displaying your folder structure as it is). From the view point - what improvements do you envision?
=> We think about supporting editing your photo library in PhotoPrism (including renaming/moving) but it will take some time as other features have a higher priority (e.g. face recognition and batch edit) 

- Metadata Editor (I guess on this PhotoPrism already exceeds all existing solutions)
 =>  :)

- Auto Update Index in case photos are moved/renamed/added/deleted outside of PhotoPrism
=> In a first step we implemented auto index/import after changes using WebDAV. We might add more logic here later. 

- Apps for Android/iOS including photo upload
=> PhotoPrism itself  is a PWA so you can add it to your homescreen already and use it with your phone. Improvements for the mobile experience are planned.
=> Apps supporting WebDAV like PhotoSync work very well to move photos from your phone to PhotoPrism automatically
=> There is a nice community app (still in beta) for android and iOS (https://github.com/photoprism/photoprism-mobile)

- Apps for the Big Screen (Android TV/Apple TV)
=> We see huge potential for PhotoPrism on big screens. So on the long run this is a potential to do. 

- User Permissions based on folders
=> Will be part of user management :)

- Don't regenerate thumbnails - use the existing once generated by Photostation (probably technically infeasible?)
=> Hard to say. It depends on what thumbnail sizes and downscaling  Photostation uses to create thumbnails. PhotoPrism depends on special thumbnail sizes e.g. for color detection or object recognition. 

I will move this to discussions so that we can keep it open for updates and as starting point for other users coming from Synology.

**Reply by GlassedSilver** [reply]
That treeview (similar to Windows Explorer or how Apple's Aperture did it) would be very welcome.

Also it makes no sense to me to have the Folders area show folders ONLY with large folder previews in the view area.

List view would be very handy in that area as well.

A no-hiearchy view (without it being an option) never strikes me as a good idea, but that seems to have been overthrown thankfully. (unless I checked the wrong place or my hierarchy in my test lib is not deep enough to stumble upon issues here :))

**Comment by wechsler42** [comment]
I would like to second @GlassedSilver with the tree view. This is something I really miss as a former PhotoStation user.

**Reply by graciousgrey** [reply]
A hierarchical version of the folder view is planned: https://github.com/photoprism/photoprism/issues/1536 :)

**Reply by GlassedSilver** [reply]
Having used Photoprism a little more the past few days I noticed it's basically missing in every place where there is a hierarchy. The scope should definitely be expanded to every section with [sub]folders, -albums and/or -filters. :)
